### PR TITLE
Bug 2142529: VM names are automatically changed to comply with RFC 1123 DNS Subdomain

### PR DESCRIPTION
**Pull request by RichardHoch**
MTV 1.7.6

Resolves https://bugzilla.redhat.com/show_bug.cgi?id=2142529 by adding note to prerequisites for all migrations.

Preview:

### CODE REVIEW

**Comment by ahadas** [comment]
@qiyuann

**Review by qiyuann — APPROVED** [review]

File: documentation/modules/source-vm-prerequisites.adoc [hunk 1]
```diff
@@ -10,6 +10,12 @@ The following prerequisites apply to all migrations:
...
-* The VM name must not duplicate the name of a VM in the {virt} environment.
 * The VM operating system must be certified and supported for use as a link:https://access.redhat.com/articles/973163#ocpvirt[guest operating system with {virt}] _and_ for link:https://access.redhat.com/articles/1351473[conversion to KVM with `virt-v2v`].
+* VM names must contain only lowercase letters (`a-z`), numbers (`0-9`), or hyphens (`-`), up to a maximum of 253 characters. The first and last characters must be alphanumeric. The name must not contain uppercase letters, spaces, periods (`.`), or special characters.
+* VM names must not duplicate the name of a VM in the {virt} environment.
++
+[NOTE]
+====
+If a VM has a name that does not comply with the rules, {project-full} automatically assigns a new name to the VM that does comply with them.
+This feature allows a migration to proceed smoothly even if someone entered a VM name that does not follow the rules.
```
Comment (bkhizgiy) [inline documentation/modules/source-vm-prerequisites.adoc:20]: lgmt just one note, maybe worth mentioning what will be the newly generated name, something like:
`The automatically generated name will be in the following format: excluded chars will be removed, upper case letters will be switched to lower case letters and all underscore ('_') will be changed to a dash ('-') instead.`

**Comment by RichardHoch** [comment]
@bkhizgiy: made slight changes to your suggestion. Please review.

File: documentation/modules/source-vm-prerequisites.adoc (hunk 1 above)
Comment (RichardHoch) [reply documentation/modules/source-vm-prerequisites.adoc:27]: Done, with slight rewrite to fit style guidelines.

Comment (bkhizgiy) [reply documentation/modules/source-vm-prerequisites.adoc:27]: ack, looking great!

**Review by bkhizgiy — APPROVED** [review]

File: documentation/modules/source-vm-prerequisites.adoc [hunk 2]
```diff
@@ -10,6 +10,19 @@ The following prerequisites apply to all migrations:
...
+* VM names must contain only lowercase letters (`a-z`), numbers (`0-9`), or hyphens (`-`), up to a maximum of 253 characters. The first and last characters must be alphanumeric. The name must not contain uppercase letters, spaces, periods (`.`), or special characters.
+* VM names must not duplicate the name of a VM in the {virt} environment.
++
+[NOTE]
+====
+If a VM has a name that does not comply with the rules, {project-full} automatically assigns a new name to the VM that does comply with them.
+
+{project-full}  makes the following changes when it automatically generates a new VM name:
+
+* Excluded charaacters are removed.
```
Comment (qiyuann) [inline documentation/modules/source-vm-prerequisites.adoc:23]: A typo here, charaacters -> characters

Comment (RichardHoch) [reply documentation/modules/source-vm-prerequisites.adoc]: Fixed.

**Review by qiyuann — APPROVED** [review]

**Pull request merged**